src/components/utils.ts

```diff
@@ -51,8 +51,13 @@ export const handleColumnChange = (column: any, setColumnValues: any, columnValu
   if (column.index === undefined) {
     column.index = lastIndex
   }
-  columnValues[column.index]['value'] = column?.value
-  setColumnValues(columnValues)
+  const newColumnValues = [
+    ...columnValues.slice(0, column.index),
```

Comment:
```suggestion
    ...columnValues.slice(0, column?.index),
```
Should there be `?` since column might be null as indicated on line 57 `value: column?.value` to keep it consistent?

Reply:
on line 51 i'm handling the case of no column.index and in which case setting it so should never be undefined

Reply:
ah i missed that